pkg/ops/rawDiskGeneration.go

```diff
@@ -231,17 +231,17 @@ func (r *RawImage) createRecoveryPartitionImage() (string, error) {
 
 	err = fsutils.MkdirAll(r.config.Fs, filepath.Join(tmpDirRecovery, "cOS"), 0755)
 
-	recoveryImage := v1.Image{
+	recoveryImage := &v1.Image{
 		File:       filepath.Join(tmpDirRecovery, "cOS", agentConstants.RecoveryImgFile),
 		FS:         agentConstants.LinuxImgFs,
 		Label:      agentConstants.SystemLabel,
 		Source:     v1.NewDirSrc(r.Source),
 		MountPoint: tmpDirRecoveryImage,
 	}
 	size, _ := config.GetSourceSize(r.config, recoveryImage.Source)
-	recoveryImage.Size = uint(size)
+	recoveryImage.Size = uint(size + 100)
```

Comment:
why is it 100 here and 150 bellow? might be worth making them constants that tell the reason for the value

Reply:
well 100 for the recovery.img and then for the partition which contains the recovery image it has to be a bit bigger as it cannot be the same size :D

Reply:
I mean, if the recovery image is 10Mb and then we also need to copy grub stuff or configs, the partition that has the image inside needs to be that size plus a bit more to account for those artifacts, it cannot be 10Mb exactly as just writing a single text file will make it not fit :D

Reply:
I'd add at least a comment with this info you mention, it's obvious after explained but I wouldn't have thought of it otherwise 😅

Reply:
added. Plus fixed the size of recovery as it needs to hold 2 images during upgrade, which I almost forgot!